CMakeLists.txt

```diff
@@ -316,17 +316,28 @@ set_var_to_condition(MIOPEN_USE_MLIR_DEFAULT NOT (NOT ${BUILD_SHARED_LIBS} AND $
 option(MIOPEN_USE_MLIR "Use MLIR compilation backend" ${MIOPEN_USE_MLIR_DEFAULT})
 
 if(MIOPEN_USE_MLIR)
-    # REQUIRED is not supported before cmake 3.18
-    find_library(LIBMLIRMIOPEN MLIRMIOpen REQUIRED)
     if(NOT ${BUILD_SHARED_LIBS} AND ${MIOPEN_USE_COMGR})
-        message(FATAL_ERROR "Potential symbol conflict between mlir and comgr in static build")
+      message(FATAL_ERROR "Potential symbol conflict between mlir and comgr in static build")
     endif()
-
-    if(NOT LIBMLIRMIOPEN)
-        message(FATAL_ERROR "libMLIRMIOpen not found, please reinstall dependencies. \
+    # Try to find package Miir
+    # REQUIRE is omitted since we do not want cmake to abort if the package is not found
+    find_package(Miir 1.0.0 CONFIG)
+    # variable MLIRMIOpen_INCLUDE_DIR is defined in a cmake file provided by the
+    # Miir package. Its definition helps us determine whether the package is found
+    if(NOT MLIRMIOpen_INCLUDE_DIR)
```

Comment:
find_package info: Regardless of the mode used, a \<PackageName\>_FOUND variable will be set to indicate whether the package was found.
\<PackageName\>_FOUND will be a more understandable solution